metrics/ci_integrations/lib/cli/command/sync_command.dart

```diff
@@ -196,6 +220,24 @@ class SyncCommand extends CiIntegrationCommand<void> with LoggerMixin {
     }
   }
 
+  /// Parses the first sync fetch limit from the given [value].
+  ///
+  /// Throws a [SyncError] if the fetch limit can't be parsed or is
+  /// not grater than `0`.
+  int parseFirstSyncFetchLimit(String value) {
+    logger.info('Parsing first sync fetch limit...');
+
+    final fetchLimit = int.tryParse(value);
+
+    if (fetchLimit == null || fetchLimit <= 0) {
```

Comment:
Would it be better to move the fetch limit validation to the `SyncConfig` class constructor?